roadmap/propagate-native-exception-through-Rust-frame.md

```diff
@@ -0,0 +1,43 @@
+# Propagate native exception through Rust frame
+
+## Summary
+
+These roadmap items refer to when a "native exception" is raised in a
+native function defined with the `"C unwind"` ABI, as might happen if
+we invoked a C++ function that throws an exception.
+
+(TODO: Investigate details of how this would be declared in any case.)
+
+In this case, the stack would look like this:
+
+```
+     ...
+      |
+[Rust frame]
+      |                            ^
+      | (calls)                    | (unwinding
+      v                            |  goes this
+[native function]                  |  way)
+      |                            |
+      +--- native function throws -+
+```
+
+We can distinguish two cases:
+
+* The Rust frame has **no in-scope destructors**.
+    * We need to define precisely when a Rust function is guaranteed to have no destructors.
```

Comment:
I'm not sure this is strictly true; I think it might actually be easier to specify the case with destructors (which is merely a superset of the case without destructors) without trying to define a "no-destructor" guarantee.

Reply:
Perhaps, yes. I guess we'll figure that out as we go. One thing that's not included in the doc is the order in which we do things per se -- we should probably write out the intended ordering, but I think it's something we should feel free to change.